examples_tests/31.SkinningDataBenchmark/1.vert

```diff
@@ -11,14 +13,22 @@ layout(std430, set = 0, binding = 0, row_major) restrict readonly buffer BoneMat
     BoneNormalMatPair matrices[];
 };
 
+#ifndef BENCHMARK
 layout(location = 0) in vec3 pos;
 layout(location = 3) in vec3 normal;
+#endif
 layout(location = 4) in uint boneID;
 
 layout(location = 0) out vec3 vNormal;
 
 void main()
 {
+    #ifdef BENCHMARK
+    const vec3 pos = vec3(1.0, 2.0, 3.0);
+    const vec3 normal = vec3(1.0, 2.0, 3.0);
+    #endif
+
     gl_Position = matrices[boneID].boneMatrix * vec4(pos, 1.0);
-    vNormal = vec3(matrices[boneID].normalMatrix * vec4(normalize(normal), 0.0));
+    //vNormal = vec3(matrices[boneID].normalMatrix * vec4(normalize(normal), 0.0));
+    vNormal = mat3(matrices[boneID].normalMatrix) * normalize(normal);

```

Comment:
do a hack `gl_Position.xyz += mat3(matrices[boneID].normalMatrix) * normalize(normal)`

Reply:
so you dont need to have an output `vNormal`

Reply:
also a cool thing  would be to ensure that `abs(gl_Position.z) > abs(gl_Position.w)` which will send the triangles outside the NDC bounds and ensure no pixels get rasterized.

an idea would be to do this
```glsl
gl_Position.w *= dot(mat3(matrices[boneID].normalMatrix)*normal,vec3(0.01));
```